src/java.desktop/unix/native/libawt_xawt/awt/awt_GraphicsEnv.c

```diff
@@ -572,6 +572,11 @@ getAllConfigs (JNIEnv *env, int screen, AwtScreenDataPtr screenDataPtr) {
 
 cleanup:
     if (success != JNI_TRUE) {
+        for (int i = 0; i < ind-1; i++) {
+            if (graphicsConfigs[i] != 0) {
```

Comment:
It it fine as it is, but this check can be omitted since `free` is no-op for null pointer arguments.